src/app/input/input.directive.ts

```diff
@@ -138,6 +154,15 @@ export class MzInputDirective extends HandlePropChanges implements OnInit {
     });
   }
 
+  setLabelActive() {
+    // need setTimeout otherwise it wont make label float in some circonstances (forcing validation for example)
+    setTimeout(() => {
+      const inputValue = (<HTMLInputElement>this.inputElement[0]).value;
+      const isActive = !!this.placeholder || !!inputValue;
+      this.renderer.setElementClass(this.labelElement[0], 'active', isActive);
```

Comment:
Meant to stay compatible with angular2 until render is really removed from angular4